File: syntax_comparison_to_python/basic_syntax.py
# A Python program that reads a variable length list of numbers from
# the command line and displays the maximum of these.
import sys

# Prompt the user for the number of elements
try:
    n = int(input('How many numbers do you want to enter? '))
except ValueError:
    print('Invalid input. Please enter a positive integer.')

    # Exit with a success status code
    sys.exit(1)

if n <= 0:
    print('Number of elements must be a positive integer.')

    # Exit with a success status code
    sys.exit(1)

# Read the numbers into a list using a loop
numbers = []
for i in range(n):
    try:
        val = float(input(f'Enter number {i+1}: '))
    except ValueError:
        print('Invalid input.')
        sys.exit(1)
    numbers.append(val)

# Compute and display the maximum
print(f'The maximum number is: {max(numbers):.2f}')

# Exit with a success status code
sys.exit(0)

File: syntax_comparison_to_python/basic_syntax.c
/* A C program that reads a variable size array of numbers from the
 * command line and displays the maximum of these.
 */
#include <stdio.h>
#include <stdlib.h>

int main(void) {
  int n; // Number of elements

  // Prompt the user for the number of elements
  printf("How many numbers do you want to enter? ");
  if (scanf("%d", &n) != 1 || n <= 0) {
    printf("Invalid input or number of elements.\n");

    // Exit with a error status code
    return 1;
  }

  // Dynamically allocate memory for 'n' doubles
  double *numbers = (double *)malloc(n * sizeof(double));
  if (numbers == NULL) {
    printf("Memory allocation failed.\n");

    // Exit with a error status code
    return 1;
  }

  // Read the numbers from the user
  for (int i = 0; i < n; i++) {
    printf("Enter number %d: ", i + 1);
    if (scanf("%lf", &numbers[i]) != 1) {
      printf("Invalid input.\n");

      // Clean up before exiting
      free(numbers);

      // Exit with a error status code
      return 1;
    }
  }

  // Find the maximum value
  double max = numbers[0];
  for (int i = 1; i < n; i++) {
    if (numbers[i] > max) {
      max = numbers[i];
    }
  }

  // Display the result
  printf("The maximum number is: %.2f\n", max);

  // Free the allocated memory
  free(numbers);

  // Successful completion
  return 0; 
}
